### PR TITLE
Docker testing

### PR DESCRIPTION
**Pull request by judgerussell**
updated dockerfiles. anybody can build and run the docker images by using
"docker-compose up -d" 
once running, you can visit the frontend at 
http://localhost:3000
and the backend at
http://localhost:8000

try it out!

### CODE REVIEW

File: docker-compose-deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+version: "3.7"
+
```
Comment (Enoch-Hsiao) [inline docker-compose-deploy.yaml]: I would separate the 2 docker-compose files into 2 separate directories (so one can cd into the appropriate directory and run `docker-compose up -d`
or include the info that in order to run "docker-compose-deploy.yaml" one must enter 
`docker-compose -f docker-compose-deploy.yaml up`

File: docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      - "8000:8000"
+
+  editor-frontend:
+    restart: always
+    command : npm start
+    container_name: front
+    build:
+      context: ./Ghost_in_the_Shell/client_folder_react_app/ethisim/
+      dockerfile: Dockerfile
+    ports:
```
Comment (Enoch-Hsiao) [inline docker-compose.yaml:25]: Not needed for this PR, but you should look into enabling hot reloading through a docker-compose file. In other words, when I, as a developer, start up the application using Docker and make a code change (usually in the front-end UI), the browser automatically updates with the change. 

I would have a separate docker-compose file for this (for developers who want hot reloading and developers who don't, as it takes up a lot of CPU resources).

File: docker-compose.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+version: "3.7"
+
+services:
+  backend:
+    restart: always
+    command: bash -c "cd lead && python manage.py runserver 0.0.0.0:8000"
+    build:
```
Comment (Enoch-Hsiao) [inline docker-compose.yaml:8]: Add a container name for clarification.

File: docker-compose-deploy.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    restart: always
+    command: bash -c "cd lead && python manage.py runserver 0.0.0.0:8000"
+    image: judgerussell/track2backend
+    ports:
+      - "8000:8000"
+
+  frontend:
+    restart: always
+    command : npm start
+    image: judgerussell/track2frontend
```
Comment (Enoch-Hsiao) [inline docker-compose-deploy.yaml:19]: Add a container names for the frontend and backend services for clarification.

**Comment by Enoch-Hsiao** [comment]
Again, not needed for PR, but it would be good to create a user-defined network in the docker-compose file to connect the 2 services (as they talk to each other). [By default Compose sets up a single network for your app ](https://docs.docker.com/compose/networking/) (so backend and frontend can talk to each other already), but in the case that we need to have more complex network topologies and isolation, it would be good to know for the future (especially for future web server configuration).

Some sources:
https://docs.docker.com/compose/networking/
https://docs.docker.com/network/
https://docs.docker.com/network/bridge/

**Comment by yiwei00** [comment]
For reference, `-d` detaches the containers so they run in the background. To close them run `docker-compose down`. If you don't want it to run in the background leave out the `-d` flag, and exit with an interrupt from your keyboard.

File: docker-compose.yaml (hunk 3 above)
Comment (Enoch-Hsiao) [reply docker-compose.yaml:8]: Still needs a container name for backend service

File: docker-compose-deploy.yaml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    container_name: frontend
+    image: judgerussell/track2frontend
+    ports:
+      - "3000:3000"
+    stdin_open: true
+    depends_on:
+      - backend
+    networks:
+      default-net
+
```
Comment (Enoch-Hsiao) [inline docker-compose-deploy.yaml:27]: services.backend.networks contains an invalid type, it should be an array, or an object
services.frontend.networks contains an invalid type, it should be an array, or an object

Think you're missing a dash, also maybe keep naming of services consistent (back vs backend, front vs frontend)

**Review by Enoch-Hsiao — APPROVED** [review]
Looks good to me!

**Pull request merged**